lib/email_spec/helpers.rb

```diff
@@ -140,9 +140,11 @@ def parse_email_for_explicit_link(email, regex)
       request_uri(url)
     end
 
+    HrefRegex = %r{href=['"]?([^'"]*)['"]?}
+    NotClosingARegex = %r{(?:(?!</a>).)*?}
     # e.g. Click here in  <a href="http://confirm">Click here</a>
     def parse_email_for_anchor_text_link(email, link_text)
-      if textify_images(email.default_part_body) =~ %r{<a[^>]*href=['"]?([^'"]*)['"]?[^>]*?>[^<]*?(.*?)#{link_text}(.*?)[^<]*?</a>}
+      if textify_images(email.default_part_body) =~ %r{<a[^>]*#{HrefRegex}[^>]*>#{NotClosingARegex}#{link_text}#{NotClosingARegex}</a>}
```

Comment:
@Hirurg103 looks good so far! could you use ruby constants for the begin and the end? I think that could clear the code up, otherwise it looks a little confusing.